### DISCUSSION TITLE
[ENH] Align Tenant.Id with Tenant.TenantId for consistency and clarity

### DISCUSSION THREAD

**Discussion by yusianglin11010** [elsa-workflows/elsa-core#6041] [Design & tech]
### Enhancement Overview
I propose a discussion on whether we should treat `Tenant.Id` as equivalent to `Tenant.TenantId`. 

The `Tenant` class, which inherits from the `Entity` class, contains a `TenantId` property. In a multi-tenancy scenario, it's straightforward for core entities to have a `TenantId` property. However, for the `Tenant` class itself, I believe that `Entity.Id` is sufficient to represent the concept of `TenantId`.

### Proposed Enhancement
I suggest overriding the `TenantId` property in the `Tenant` class as follows:

```csharp
/// <summary>
/// TenantId is the same as Entity ID.
/// </summary>
public override string TenantId => Id;
```

**Description of the enhancement**
This change could improve readability and prevent potential misuse of the `Tenant` class. It ensures conceptual consistency, as a Tenant's `TenantId` is logically identical to its `Id`. Additionally, it avoids redundancy by preventing the need to manage two identical values.

### Impact of Enhancement
The proposed solution would eliminate the flexibility to set a Tenant's `TenantId` and `Id` to different values. However, I haven't identified any specific use cases where this separate assignment would be necessary, and allowing it might introduce ambiguity.

### Additional Context
I encountered an error while running the [sample project](https://github.com/elsa-workflows/elsa-samples/blob/main/src/aspnet/Elsa.Samples.AspNet.Tenants/)  to this line

https://github.com/elsa-workflows/elsa-core/blob/5cb7cbf66d08a25601e1296872220e9627efeaed/src/modules/Elsa.Tenants/Services/DefaultTenantResolverPipelineInvoker.cs#L20

Because the app injected the tenant list from a config file with `TenantId` configured but lacking `Id`, then cause an error while trying to construct a dictionary with missing `Tenant.Id`. Making `Tenant.TenantId` programmatically identical to `Tenant.Id` could improve the clarity of its usage and prevent similar unexpected errors.

I'm open to hearing any considerations I might have overlooked and welcome your opinions on this matter.

**Comment by sfmskywalker** [comment]
I was actually thinking that a Tenant's TenantId doesn't represent its own primary key, but the tenant it is itself a part of. In other words, a tenant can itself be part of a parent tenant.

**Comment by yusianglin11010** [comment]
Thanks for the quick reply and great explanation about hierarchical tenants. That makes a lot of sense now.
I think we've covered the main point here. Would you prefer to convert this to a discussion, or should we just close it? Either way works for me.

**Comment by sfmskywalker** [comment]
Great! Good idea, let's convert this to a discussion 👍🏻

**Comment by jdevillard** [comment]
What about ParentTenantId ? to avoid confusion?
